lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,180 @@
+// @flow
+/**
+ *
+ * Typeahead
+ *
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import classNames from 'classnames';
+
+import styles from './Typeahead.style';
+import type { Props } from './types';
+import Input from '../../atoms/Input';
+import List from '../List';
+import Label from '../../atoms/Label';
+import ListItem from './ListItem';
+import { ENTER_KEY, UP_ARROW, DOWN_ARROW, ESCAPE, TAB } from './constants';
+
+const Typeahead = ({
+  id,
+  getResults,
+  results,
+  className,
+  renderSuggestionItem,
+  searchValue,
+  onChangeHandler,
+  getSelectedSuggestionValue,
+  label,
+  isFieldLabel,
+  ...otherProps
+}: Props): React.Element<any> => {
+  const [updatedResults, setFilteredResults] = useState(results);
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [showResults, setShowResults] = useState(true);
+
+  const triggerGetResults = val => {
+    return val && val.trim().length >= 3;
+  };
+
+  /**
+   * Function to handle item selection
+   * @param {'*'} e
+   */
+  const handleClick = item => {
+    setActiveSuggestion(0);
+    getSelectedSuggestionValue(item);
+    setShowResults(false);
+  };
+
+  const getFilteredResults = (allResults, inputText) => {
+    const filteredResults = allResults.filter(
+      suggestion => suggestion.toLowerCase().indexOf(inputText.toLowerCase()) > -1
```

Comment:
Good to Have: 

See if https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare will be a better fit, may be as a flagged feature